test/unittest_suite.py

```diff
@@ -1,24 +1,37 @@
 #!/usr/bin/env python
 
 import sys
+
 import unittest2 as unittest
 
+
 if sys.version_info[0] == 2:
     modules = [
         'test_katello.test_enabled_report',
         'test_katello.test_packages',
         'test_katello.test_repos',
         'test_katello.test_utils',
-        'test_rhsm_fact_plugin',
-        'test_yum_plugins.test_enabled_repos_upload',
-        'zypper_plugins.test_enabled_repos_upload',
-        'zypper_plugins.test_tracer_upload'
+        'test_rhsm_fact_plugin'
     ]
 
+    try:
+        import zypp_plugin
+        zypper_enabled = True
+    except ImportError:
+        zypper_enabled = False
+
+    if zypper_enabled:
+        modules.append('zypper_plugins.test_enabled_repos_upload')
+        modules.append('zypper_plugins.test_package_upload')
+        modules.append('zypper_plugins.test_tracer_upload')
+    else:
+        modules.append('test_yum_plugins.test_enabled_repos_upload')
+
     if sys.version_info[1] > 6:
-        modules.append('test_katello.test_plugin')
-        modules.append('test_katello.test_agent.test_pulp.test_handler')
-        modules.append('test_katello.test_agent.test_pulp.test_libyum')
+        if not zypper_enabled:
```

Comment:
wouldn't this work?
```python
    if sys.version_info[1] > 6 and not zypper_enabled:
        modules.append('test_katello.test_plugin')
        modules.append('test_katello.test_agent.test_pulp.test_handler')
        modules.append('test_katello.test_agent.test_pulp.test_libyum')
    else:
        # this test file doesn't start with test_ to avoid py3 unittest discovery
        modules.append('test_katello.legacy_plugin_test')
```

Reply:
I had actually written it that way but it bit me! The else clause is exclusive to the `sys.version_info[1] > 6` expression because if it evals true (We are on python 2.7+) and `not zypper_enabled` evals false in the single expression example then the `else` clause would be reached and so the legacy_plugin_test would be added to the run for python 2.7+ which would fail

Reply:
@johnpmitsch I updated the code and nested this if block within the above `else` which makes it more readable and the result is the same. check it out

Reply:
that makes sense, looks good! Thanks for the explanation